test/unit-tests/otcWrapperV2.test.ts

```diff
@@ -0,0 +1,2038 @@
+      // resulting address = 0xa94ab2bb0c67842fb40a1068068df1225a031a7d
+      const randomBuffer = Buffer.alloc(32, 'dsaas')
+
+      const userWallet = Wallet.fromPrivateKey(randomBuffer)
+
+      const owner = userWallet.getAddressString()
+      const value = parseUnits('5000', 6).toNumber()
+      const nonce = 0
+      const spender = otcWrapperProxy.address
+      const maxDeadline = new BigNumber(await time.latest()).plus(8600000 * 50).toString()
+
+      // fund eth
+      forceSend = await ForceSend.new(addressBook.address)
+      await forceSend.go(owner, { value: ethers.utils.parseEther('2').toString() })
+
+      // fund usdc
+      await usdc.mint(owner, createTokenAmount(200000, USDCDECIMALS))
+
+      const buildData = (chainId: number, verifyingContract: string, deadline = maxDeadline) => ({
+        primaryType: 'Permit',
+        types: { EIP712Domain, Permit },
+        domain: { name, version, chainId, verifyingContract },
+        message: { owner, spender, value, nonce, deadline },
+      })
+
+      const data = buildData((await usdc.getChainId()).toNumber(), usdc.address)
+      const signature = ethSigUtil.signTypedMessage(userWallet.getPrivateKey(), { data })
+      const { v, r, s } = fromRpcSig(signature)
+
+      const acct = owner
+      const amount = value.toString()
+      const deadline = maxDeadline
+
+      userSignature1 = { amount, deadline, acct, v, r, s }
+    })
+    it('set up user permit USDC signature 2', async () => {
+      // resulting address = 0xa94ab2bb0c67842fb40a1068068df1225a031a7d
+      const randomBuffer = Buffer.alloc(32, 'dsaas')
+
+      const userWallet = Wallet.fromPrivateKey(randomBuffer)
+
+      const owner = userWallet.getAddressString()
+      const value = parseUnits('5000', 6).toNumber()
+      const nonce = 1
+      const spender = otcWrapperProxy.address
+      const maxDeadline = new BigNumber(await time.latest()).plus(8600000 * 50).toString()
+
+      // fund eth
+      forceSend = await ForceSend.new(addressBook.address)
+      await forceSend.go(owner, { value: ethers.utils.parseEther('2').toString() })
+
+      // fund usdc
+      await usdc.mint(owner, createTokenAmount(200000, USDCDECIMALS))
+
+      const buildData = (chainId: number, verifyingContract: string, deadline = maxDeadline) => ({
+        primaryType: 'Permit',
+        types: { EIP712Domain, Permit },
+        domain: { name, version, chainId, verifyingContract },
+        message: { owner, spender, value, nonce, deadline },
+      })
+
+      const data = buildData((await usdc.getChainId()).toNumber(), usdc.address)
+      const signature = ethSigUtil.signTypedMessage(userWallet.getPrivateKey(), { data })
+      const { v, r, s } = fromRpcSig(signature)
+
+      const acct = owner
+      const amount = value.toString()
+      const deadline = maxDeadline
+
+      userSignature2 = { amount, deadline, acct, v, r, s }
+    })
+    it('set up user permit USDC signature 3', async () => {
```

Comment:
can we consolidate all these in init (since these aren't really unit tests) + do some sort of iteration